streaming/rdm_atomic.c

```diff
@@ -405,27 +405,27 @@ static int alloc_ep_res(struct fi_info *fi)
 	// registers local data buffer buff that specifies
 	// the first operand of the atomic operation
 	ret = fi_mr_reg(domain, buf, buf_size,
-		FI_REMOTE_READ | FI_REMOTE_WRITE, 0,
-		get_mr_key(), 0, &mr, NULL);
+			(FI_SEND | FI_RECV | FI_READ | FI_WRITE |
+			 FI_REMOTE_READ | FI_REMOTE_WRITE), 0,
```

Comment:
Only set FI_SEND etc if FI_MR_LOCAL is required by the provider.  This helps uncover provider bugs.